relational_operators/WorkOrderFactory.cpp

```diff
@@ -453,6 +455,9 @@ bool WorkOrderFactory::ProtoIsValid(const serialization::WorkOrder &proto,
 
       return proto.HasExtension(serialization::BuildHashWorkOrder::any_join_key_attributes_nullable) &&
              proto.HasExtension(serialization::BuildHashWorkOrder::block_id) &&
+             proto.HasExtension(serialization::BuildHashWorkOrder::join_hash_table_group_index) &&
+             query_context.isValidJoinHashTableId(
+                 proto.GetExtension(serialization::BuildHashWorkOrder::join_hash_table_group_index)) &&
              proto.HasExtension(serialization::BuildHashWorkOrder::join_hash_table_index) &&
              query_context.isValidJoinHashTableId(
                  proto.GetExtension(serialization::BuildHashWorkOrder::join_hash_table_index));
```

Comment:
We should remove this condition check, and combine it to the check for both the group id and the local id within the group.